runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/WindmillSink.java

```diff
@@ -215,12 +219,12 @@ public long add(WindowedValue<T> data) throws IOException {
         productionMap.put(key, keyedOutput);
       }
 
-      Windmill.Message.Builder builder =
-          Windmill.Message.newBuilder()
-              .setTimestamp(WindmillTimeUtils.harnessToWindmillTimestamp(data.getTimestamp()))
-              .setData(value)
-              .setMetadata(metadata);
-      keyedOutput.addMessages(builder.build());
+      messageBuilder.clear();
+      messageBuilder
+          .setTimestamp(WindmillTimeUtils.harnessToWindmillTimestamp(data.getTimestamp()))
+          .setData(value)
+          .setMetadata(metadata);
+      keyedOutput.addMessages(messageBuilder.build());
```

Comment:
seems better to clear after use (perhaps in try/finally to handle exceptions) so that we don't keep reference to the memory used